google-services-plugin/build.gradle.kts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2015 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+plugins {
+    id("java-gradle-plugin")
+    id("org.jetbrains.kotlin.jvm") version "1.7.22"
```

Comment:
If we're migrating to Kotlin, might as well migrate to the Kotlin DSL as well :)

Reply:
do you mean use the ``kotlin-dsl`` plugin here?
AFAIK it does more things than required and binds you to an outdated Kotlin version (although I might be wrong o that). What I can do is try to add a `dependency` on just the Kotlin DSL helpers from Gradle and see if it makes the plugin code more readable in Kotlin